### PR TITLE
Added brainfuck code and explanation

### PR DESCRIPTION
**Pull request by Ariana1729**
Just so that multiple algos and langs dont get merged, there's another PR

### CODE REVIEW

**Comment by Ariana1729** [comment]
ok umm not quite sure how to fix the conflicts tbh ill try again maybe like tmr?

**Comment by leios** [comment]
I'll fix the conflicts once the code is reviewed. These are easy ones to fix on my end.

**Comment by leios** [comment]
I think @Trashtalk217 did a decent BF review last time. Can he also do this review?

File: contents/euclidean_algorithm/code/brainfuck/euclidean_sub.bf [hunk 1]
```diff
@@ -0,0 +1,2 @@
+,>,>>>+[[-]<<<<[->>>+<<<]>[->>>+<<<]>>[-<+<<+>>>]>[-<+<<+>>>]<<[->->[-]+<[>-]>[->>]<<<]>[>]<[<<[-]>>[-<<+>>>+<]]<[<<[-]>>[-<<+>>>>+<<]]>>>[-]+>[-]<<[>-]>[<<<<.>>>>->>]<<]
```
Comment (Trashtalk217) [inline contents/euclidean_algorithm/code/brainfuck/euclidean_sub.bf]: I'd first like to note it's really hard commenting on a onliner but I'll do my best: the second `[-]` is unnecessary, I've never seen it called.

File: book.json [hunk 2]
```diff
@@ -163,6 +163,10 @@
         {
           "lang": "lolcode",
           "name": "LOLCODE"
+        },
+        {
+          "lang": "bf",
+          "name": "Brainfuck"
```
[the remark this answers is not in the document]
Comment (Trashtalk217) [reply book.json]: Ignore this, Its double

File: contents/euclidean_algorithm/code/brainfuck/euclidean_sub.bf (hunk 1 above)
Comment (Trashtalk217) [inline contents/euclidean_algorithm/code/brainfuck/euclidean_sub.bf]: I'm just going to leave a nice indented version down below, so I can accuratly point some things out.
````
,>,

>>>+

[
	[-]
	<<<<
	[->>>+<<<]
	>
	[->>>+<<<]
	>>
	[-<+<<+>>>]
	>
	[-<+<<+>>>]
	<<
	[
		->->
		**[-]**
		+<
		[>-]
		>
		[->>]
		<<<
	]
	>
	[>]
	<
	[
		<<
		[-]
		>>
		[-<<+>>>+<]
	]
	<
	[
		<<
		[-]
		>>
		[-<<+>>>>+<<]
	]
	>>>
	**[-]**
	+**>
	[-]
	<**<
	[>-]
	>
	[<<<<.>>>>->>]
	<<
]	
````
The parts I surrounded with ** I found to be unneeded. And I have been able to leave them away without much issue.

File: contents/euclidean_algorithm/code/brainfuck/euclidean_mod.bf [hunk 3]
```diff
@@ -0,0 +1 @@
+>,>,[<[>->+<[>]>[<+>-]<<[<]>-]>[-<+>]>[-<+<+>>]<]<.
```
Comment (Trashtalk217) [inline contents/euclidean_algorithm/code/brainfuck/euclidean_mod.bf]: I'm generally happy with this code, it did take some checking to find out that the `[>->+<[>]>[<+>-]<<[<]>-]` part, you got from [here](https://esolangs.org/wiki/Brainfuck_algorithms#Modulus_algorithm) was in the public domain, but I eventually found [this](https://esolangs.org/wiki/Esolang:Policy), so it's fine.

**Comment by Ariana1729** [comment]
oh yes hi, updated the code&updated this branch

File: contents/euclidean_algorithm/euclidean_algorithm.md [hunk 4]
```diff
@@ -209,6 +213,114 @@ and modulo method:
 [import, lang="LOLCODE"](code/lolcode/euclid.lol)
 {% sample lang="bash" %}
 [import, lang="bash"](code/bash/euclid.bash)
+{% sample lang="bf" %}
+#### Subtraction varient
+##### Code
```
Comment (Trashtalk217) [inline contents/euclidean_algorithm/euclidean_algorithm.md]: I'm happy with the code at this point. The code is as slim as it can be without having to redo the fundemantals. This leaves me with one final question, and that one is of style: What is your justification for writing your code as a one-liner and then a explanation in the md-file? We already have a brainfuck bubble sort and it is quiet different -> [here](https://www.algorithm-archive.org/contents/bubble_sort/bubble_sort.html)

Comment (Ariana1729) [reply contents/euclidean_algorithm/euclidean_algorithm.md]: iirc there was a discussion in the discord chat about if to combine the explanation and code, and we agreed on removing all comments in the code to make it more 'brainfuck', i see if i can find that chat

Comment (Ariana1729) [reply contents/euclidean_algorithm/euclidean_algorithm.md]: https://github.com/algorithm-archivists/algorithm-archive/pull/407#issuecomment-426151437

Comment (jiegillet) [reply contents/euclidean_algorithm/euclidean_algorithm.md]: I used the "source code" then commented version with my whitespace submission. I think it's best this way. Brainfuck is a joke language and the joke is more effective without comments, but then after you see the source code you want to know more and that's when the comments are nice.

Comment (Trashtalk217) [reply contents/euclidean_algorithm/euclidean_algorithm.md]: I just find the format we already have for the bubble sort bf more easy to read and review and it also keeps documentation and code in the same place, which I prefer. And when this PR is added we'll lose some concistency between chapters. Still, having a style guide for brainfuck code seams a bit ridiculous and if this is the way some other people already agreed on, I'm fine with that. In that case I will start reviewing the explanation part.

Comment (Ariana1729) [reply contents/euclidean_algorithm/euclidean_algorithm.md]: wait so do we want like 
`Brainfuck is a joke language and the joke is more effective without comments` & reformat prev code
or like same as previous pr

Comment (Trashtalk217) [reply contents/euclidean_algorithm/euclidean_algorithm.md]: @jiegillet prefers the first and I prefer the latter. I've asked Leios what he thinks of it, he'll look at it later.

Comment (leios) [reply contents/euclidean_algorithm/euclidean_algorithm.md]: I understand that the in-text code should be more BF-like; however, a commented version is also necessary (if possible). Both this chapter and [bubble sort](https://www.algorithm-archive.org/contents/bubble_sort/bubble_sort.html) provide an adequate solution; however, I would argue that the bubble sort seems much easier to read when the two chapters are put side-by-side.

I think this ultimately boils down to a number of factors:
1. a single line of BF code does not actually fit on the page when rendering online
2. the explanation provided for this chapter is a bit hard to follow. Somehow the commented Bubble sort implementation is easier, even though it has less detail. 

With other esoteric languages (like [whitespace](https://www.algorithm-archive.org/contents/bubble_sort/bubble_sort.html)), the code was displayed as it would be seen in actually whitespace code, and then a heavily commented / modified version was put underneath. I think this is also a fine solution, but if we do that we need to make sure that the BF code actually fits into the AAA codebox.

I think the best solution is just to provide a heavily commented BF code sample with the explanation in-built. If that's not possible, the explanation at the end of the chapter is still useful, but should be rewritten (which I can help with). Sorry for all the extra work.

**Comment by Ariana1729** [comment]
was kinda busy with like music and school so was afk the whole of nov:/
Not quite sure how well this explanation is but at least its slightly better

File: contents/euclidean_algorithm/euclidean_algorithm.md [hunk 5]
```diff
@@ -209,6 +213,131 @@ and modulo method:
...
+```
+[-<+<<+>>>]
+>[-<+<<+>>>]
+```
+State: `a b a b (0) 0 0`
+
+**Here, we subtract 1 from cell 2 from 3 until one of them hits 0, the other cell would be |a-b|**
+```
+<<
+[->- subtracts a from b, assuming a>b
```
Comment (benchislett) [inline contents/euclidean_algorithm/euclidean_algorithm.md]: Wouldn't `a>b` actually affect the code execution here?

Comment (Ariana1729) [reply contents/euclidean_algorithm/euclidean_algorithm.md]: yeah that’s one of the reasons why I didn’t comment in the actual bf code, cuz .,<> are all affecting execution
